server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterStateDiffIT.java

```diff
@@ -570,9 +570,7 @@ public IndexMetadata randomChange(IndexMetadata part) {
                         }
                         break;
                     case 2:
-                        builder.settings(
-                            Settings.builder().put(part.getSettings()).put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
-                        );
+                        builder.settings(Settings.builder().put(part.getSettings()).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0));
```

Comment:
Why do we need this change?

Reply:
Changing the UUID of an index isn't possible in real life, but this test does change it and
it fails validation logic that was added in `Metadata.Builder.removeAlias()` method.

In this method we keep track of aliased indices by `Index` and that contains the uuid.
The exception on line 1359 is thrown. Basically because during this test, we (randomly) 
add an index with alias, change the uuid setting and then remove the alias. An entry
is attempted be removed, but doesn't exist, because uuid changed.

Reply:
Ah got it, so our validation actually got better here I guess :)

Reply:
I guess so :) 

All of our settings validation occurs when providing settings updates via the api. In tests, like here all the settings validation is by-passed, because we directly work with `Settings` class.